Unable to build custom nodes from n8n-nodes-starter due to eslint 404

**Describe the bug**
When using npm link to connect the n8n global install and the n8n-nodes-start project, you receive a 404. The workflow is described in this stream from n8n: https://www.youtube.com/watch?v=OI6zHJ56eW0 from roughly 4-9 minutes. It looks like this was brought about in PR 4050 (https://github.com/n8n-io/n8n/pull/4050)

npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@n8n_io%2feslint-config - Not found
npm ERR! 404 
npm ERR! 404  '@n8n_io/eslint-config@latest' is not in this registry.


**To Reproduce**
Steps to reproduce the behavior:

1. Clone the n8n-nodes-starter project
2. Install n8n globally (npm install n8n -g)
3. From the starter project directory, run (npm link)
4. From your global install directory, ie /usr/local/lib/node_modules, go to the n8n subdirectory and link your starter project (npm link n8n-nodes-<yourprojectname>

**Expected behavior**
The link appears to be referencing a package which has not been published (@n8n_io/eslint-config@latest). On the n8n project page (https://www.npmjs.com/package/n8n) the eslint project is listed, but it directs to a 404. 

**Environment (please complete the following information):**

- OS: MacOS 12.5.1
- n8n 0.194.0
- Node.js Version 18.9.0
- Database system SQLite
- Operation mode own

Just did a fresh local install and I can't install modules from the cli

```bash
~ on  main [?] 
❯ npm uninstall n8n --location=global

~ on  main [?] 
❯ npm install n8n --location=global  
npm WARN deprecated request-promise-native@1.0.9: request-promise-native has been deprecated because it extends the now deprecated request package, see https://github.com/request/request/issues/3142
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated formidable@1.2.6: Please upgrade to latest, formidable@v2 or formidable@v3! Check these notes: https://bit.ly/2ZEqIau
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSearchParams API instead.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
npm WARN deprecated core-js@2.6.12: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.

added 1139 packages, and audited 1140 packages in 45s

106 packages are looking for funding
  run `npm fund` for details

5 high severity vulnerabilities

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.

~ on  main [?] 
❯ cd /Users/test/.nvm/versions/node/v16.15.1/lib/node_modules/n8n

lib/node_modules/n8n on  main [?] is 📦 v0.194.0 via  v16.15.1 
❯ npm install n8n-nodes-starter            
npm ERR! code E404
npm ERR! 404 Not Found - GET https://registry.npmjs.org/@n8n_io%2feslint-config - Not found
npm ERR! 404 
npm ERR! 404  '@n8n_io/eslint-config@latest' is not in this registry.
npm ERR! 404 
npm ERR! 404 Note that you can also install from a
npm ERR! 404 tarball, folder, http url, or git url.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/test/.npm/_logs/2022-09-20T11_04_32_755Z-debug-0.log
```

--- a/packages/node-dev/commands/build.ts
+++ b/packages/node-dev/commands/build.ts
@@ -44,7 +44,7 @@ export class Build extends Command {
 
 			const outputDirectory = await buildFiles(options);
 
-			this.log(`The nodes got build and saved into the following folder:\n${outputDirectory}`);
+			this.log(`The nodes got built and saved into the following folder:\n${outputDirectory}`);
 		} catch (error) {
 			// eslint-disable-next-line @typescript-eslint/restrict-template-expressions, @typescript-eslint/no-unsafe-member-access
 			this.log(`\nGOT ERROR: "${error.message}"`);

--- a/packages/node-dev/package.json
+++ b/packages/node-dev/package.json
@@ -48,7 +48,6 @@
   "devDependencies": {
     "@n8n_io/eslint-config": "",
     "@oclif/dev-cli": "^1.22.2",
-    "@types/copyfiles": "^2.1.1",
     "@types/express": "^4.17.6",
     "@types/inquirer": "^6.5.0",
     "@types/node": "^16.11.22",
@@ -59,7 +58,7 @@
     "@oclif/command": "^1.5.18",
     "@oclif/errors": "^1.2.2",
     "change-case": "^4.1.1",
-    "copyfiles": "^2.1.1",
+    "fast-glob": "^3.2.5",
     "inquirer": "^7.0.1",
     "n8n-core": "~0.134.0",
     "n8n-workflow": "~0.116.0",

--- a/packages/node-dev/src/Build.ts
+++ b/packages/node-dev/src/Build.ts
@@ -2,7 +2,7 @@
 /* eslint-disable @typescript-eslint/no-unsafe-member-access */
 import glob from 'fast-glob';
 import { spawn } from 'child_process';
-import { copyFile, readFile, writeFile } from 'fs/promises';
+import { copyFile, mkdir, readFile, writeFile } from 'fs/promises';
 import { join, dirname, resolve as resolvePath } from 'path';
 import { file as tmpFile } from 'tmp-promise';
 
@@ -50,15 +50,25 @@ export async function createCustomTsconfig() {
  * @returns {Promise<string>}
  */
 export async function buildFiles(options?: IBuildOptions): Promise<string> {
-	// eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, no-param-reassign
-	options = options || {};
+	options = options ?? {};
 
 	const tscPath = join(dirname(require.resolve('typescript')), 'tsc');
 	const tsconfigData = await createCustomTsconfig();
 
 	const outputDirectory =
-		// eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-		options.destinationFolder || UserSettings.getUserN8nFolderCustomExtensionPath();
+		options.destinationFolder ?? UserSettings.getUserN8nFolderCustomExtensionPath();
+
+	await Promise.all(
+		['*.svg', '*.png', '*.node.json'].map(async (filenamePattern) => {
+			const files = await glob(`**/${filenamePattern}`);
+			for (const file of files) {
+				const src = resolvePath(process.cwd(), file);
+				const dest = resolvePath(outputDirectory, file);
+				await mkdir(dirname(dest), { recursive: true });
+				await copyFile(src, dest);
+			}
+		}),
+	);
 
 	// Supply a node base path so that it finds n8n-core and n8n-workflow
 	const nodeModulesPath = join(__dirname, '../../node_modules/');
@@ -103,17 +113,6 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 		// Remove the tmp tsconfig file
 		await tsconfigData.cleanup();
 	}
-
-	await Promise.all(
-		['*.svg', '*.png', '*.node.json'].map(async (filenamePattern) => {
-			const files = await glob(`**/${filenamePattern}`);
-			for (const file of files) {
-				const src = resolvePath(process.cwd(), file);
-				const dest = resolvePath(outputDirectory, file);
-				await copyFile(src, dest);
-			}
-		}),
-	);
 
 	return outputDirectory;
 }

--- a/packages/node-dev/src/Build.ts
+++ b/packages/node-dev/src/Build.ts
@@ -1,40 +1,26 @@
 /* eslint-disable @typescript-eslint/no-unsafe-argument */
 /* eslint-disable @typescript-eslint/no-unsafe-member-access */
-import { ChildProcess, spawn } from 'child_process';
-
-import { readFile as fsReadFile } from 'fs/promises';
-import { write as fsWrite } from 'fs';
-
-import { join } from 'path';
-import { file } from 'tmp-promise';
-import { promisify } from 'util';
+import glob from 'fast-glob';
+import { spawn } from 'child_process';
+import { copyFile, readFile, writeFile } from 'fs/promises';
+import { join, dirname, resolve as resolvePath } from 'path';
+import { file as tmpFile } from 'tmp-promise';
 
 import { UserSettings } from 'n8n-core';
-// eslint-disable-next-line import/no-cycle
 import { IBuildOptions } from '.';
-
-// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-var-requires
-const copyfiles = require('copyfiles');
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const fsReadFileAsync = promisify(fsReadFile);
-const fsWriteAsync = promisify(fsWrite);
 
 /**
  * Create a custom tsconfig file as tsc currently has no way to define a base
  * directory:
  * https://github.com/Microsoft/TypeScript/issues/25430
- *
- * @export
- * @returns
  */
 // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
 export async function createCustomTsconfig() {
 	// Get path to simple tsconfig file which should be used for build
-	const tsconfigPath = join(__dirname, '../../src/tsconfig-build.json');
+	const tsconfigPath = join(dirname(require.resolve('n8n-node-dev/src')), 'tsconfig-build.json');
 
 	// Read the tsconfig file
-	const tsConfigString = await fsReadFile(tsconfigPath, { encoding: 'utf8' });
+	const tsConfigString = await readFile(tsconfigPath, { encoding: 'utf8' });
 	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
 	const tsConfig = JSON.parse(tsConfigString);
 
@@ -47,10 +33,8 @@ export async function createCustomTsconfig() {
 	tsConfig.include = newIncludeFiles;
 
 	// Write new custom tsconfig file
-	// eslint-disable-next-line @typescript-eslint/no-unsafe-call
-	// eslint-disable-next-line @typescript-eslint/unbound-method
-	const { fd, path, cleanup } = await file();
-	await fsWriteAsync(fd, Buffer.from(JSON.stringify(tsConfig, null, 2), 'utf8'));
+	const { path, cleanup } = await tmpFile();
+	await writeFile(path, JSON.stringify(tsConfig, null, 2));
 
 	return {
 		path,
@@ -69,16 +53,7 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 	// eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing, no-param-reassign
 	options = options || {};
 
-	let typescriptPath;
-
-	// Check for OS to designate correct tsc path
-	if (process.platform === 'win32') {
-		typescriptPath = '../../node_modules/TypeScript/lib/tsc';
-	} else {
-		typescriptPath = '../../node_modules/.bin/tsc';
-	}
-	const tscPath = join(__dirname, typescriptPath);
-
+	const tscPath = join(dirname(require.resolve('typescript')), 'tsc');
 	const tsconfigData = await createCustomTsconfig();
 
 	const outputDirectory =
@@ -94,9 +69,8 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 		buildCommand += ' --watch';
 	}
 
-	let buildProcess: ChildProcess;
 	try {
-		buildProcess = spawn('node', buildCommand.split(' '), {
+		const buildProcess = spawn('node', buildCommand.split(' '), {
 			windowsVerbatimArguments: true,
 			cwd: process.cwd(),
 		});
@@ -110,8 +84,10 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 
 		// Make sure that the child process gets also always terminated
 		// when the main process does
-		process.on('exit', () => {
-			buildProcess.kill();
+		process.on('exit', () => buildProcess.kill());
+
+		await new Promise<void>((resolve) => {
+			buildProcess.on('exit', () => resolve());
 		});
 	} catch (error) {
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
@@ -122,26 +98,22 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 			errorMessage = `${errorMessage}\nGot following output:\n${error.stdout}`;
 		}
 
-		// Remove the tmp tsconfig file
-		// eslint-disable-next-line @typescript-eslint/no-floating-promises
-		tsconfigData.cleanup();
-
 		throw new Error(errorMessage);
+	} finally {
+		// Remove the tmp tsconfig file
+		await tsconfigData.cleanup();
 	}
 
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
-	return new Promise((resolve, reject) => {
-		['*.png', '*.node.json'].forEach((filenamePattern) => {
-			// eslint-disable-next-line @typescript-eslint/no-unsafe-call
-			copyfiles([join(process.cwd(), `./${filenamePattern}`), outputDirectory], { up: true }, () =>
-				resolve(outputDirectory),
-			);
-		});
-		// eslint-disable-next-line @typescript-eslint/no-unused-vars
-		buildProcess.on('exit', (code) => {
-			// Remove the tmp tsconfig file
-			// eslint-disable-next-line @typescript-eslint/no-floating-promises
-			tsconfigData.cleanup();
-		});
-	});
+	await Promise.all(
+		['*.png', '*.node.json'].map(async (filenamePattern) => {
+			const files = await glob(`**/${filenamePattern}`);
+			for (const file of files) {
+				const src = resolvePath(process.cwd(), file);
+				const dest = resolvePath(outputDirectory, file);
+				await copyFile(src, dest);
+			}
+		}),
+	);
+
+	return outputDirectory;
 }

--- a/packages/node-dev/src/Build.ts
+++ b/packages/node-dev/src/Build.ts
@@ -105,7 +105,7 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 	}
 
 	await Promise.all(
-		['*.png', '*.node.json'].map(async (filenamePattern) => {
+		['*.svg', '*.png', '*.node.json'].map(async (filenamePattern) => {
 			const files = await glob(`**/${filenamePattern}`);
 			for (const file of files) {
 				const src = resolvePath(process.cwd(), file);

--- a/packages/node-dev/src/Build.ts
+++ b/packages/node-dev/src/Build.ts
@@ -44,26 +44,20 @@ export async function createCustomTsconfig() {
 
 /**
  * Builds and copies credentials and nodes
- *
- * @export
- * @param {IBuildOptions} [options] Options to overwrite default behaviour
- * @returns {Promise<string>}
  */
-export async function buildFiles(options?: IBuildOptions): Promise<string> {
-	options = options ?? {};
-
+export async function buildFiles({
+	destinationFolder = UserSettings.getUserN8nFolderCustomExtensionPath(),
+	watch,
+}: IBuildOptions): Promise<string> {
 	const tscPath = join(dirname(require.resolve('typescript')), 'tsc');
 	const tsconfigData = await createCustomTsconfig();
-
-	const outputDirectory =
-		options.destinationFolder ?? UserSettings.getUserN8nFolderCustomExtensionPath();
 
 	await Promise.all(
 		['*.svg', '*.png', '*.node.json'].map(async (filenamePattern) => {
 			const files = await glob(`**/${filenamePattern}`);
 			for (const file of files) {
 				const src = resolvePath(process.cwd(), file);
-				const dest = resolvePath(outputDirectory, file);
+				const dest = resolvePath(destinationFolder, file);
 				await mkdir(dirname(dest), { recursive: true });
 				await copyFile(src, dest);
 			}
@@ -74,8 +68,8 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 	const nodeModulesPath = join(__dirname, '../../node_modules/');
 	let buildCommand = `${tscPath} --p ${
 		tsconfigData.path
-	} --outDir ${outputDirectory} --rootDir ${process.cwd()} --baseUrl ${nodeModulesPath}`;
-	if (options.watch === true) {
+	} --outDir ${destinationFolder} --rootDir ${process.cwd()} --baseUrl ${nodeModulesPath}`;
+	if (watch) {
 		buildCommand += ' --watch';
 	}
 
@@ -97,7 +91,7 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 		process.on('exit', () => buildProcess.kill());
 
 		await new Promise<void>((resolve) => {
-			buildProcess.on('exit', () => resolve());
+			buildProcess.on('exit', resolve);
 		});
 	} catch (error) {
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
@@ -114,5 +108,5 @@ export async function buildFiles(options?: IBuildOptions): Promise<string> {
 		await tsconfigData.cleanup();
 	}
 
-	return outputDirectory;
+	return destinationFolder;
 }

--- a/packages/node-dev/src/tsconfig-build.json
+++ b/packages/node-dev/src/tsconfig-build.json
@@ -1,10 +1,18 @@
 {
-	"extends": "../../tsconfig.json",
 	"compilerOptions": {
+		"strict": true,
+		"module": "commonjs",
+		"moduleResolution": "node",
+		"target": "es2019",
+		"lib": ["es2019", "es2020"],
 		"importHelpers": true,
+		"esModuleInterop": true,
+		"declaration": true,
+		"incremental": true,
+		"sourceMap": true
 	},
 	"include": [
-		"*.credentials.ts",
-		"*.node.ts"
+		"**/*.credentials.ts",
+		"**/*.node.ts"
 	]
 }

